Sources/GRPC/Server.swift

```diff
@@ -451,6 +454,22 @@ extension Server {
   }
 }
 
+extension Server.Configuration {
+  /// Configuration for extracting IDs from metadata and inserting them into a logger.
+  public struct TraceIDExtractor {
```

Comment:
Is this sufficiently flexible? Do we need to allow the user to define a type that can do arbitrary searches?

Reply:
Good question. I suspect not, I can imagine a situation where we could fallback to a different header field for example.

Reply:
Whoa wait a moment guys -- this is reinventing tracing infrastructure and types. Such extract/inject things are exactly what a Tracer (or other Instrument (a Tracer is-a Instrument)) would be doing. So rather, grpc should instrument itself using the tracing instruments, and make use of those to populate baggage, which then is fed into logging.

Here's the basic API: https://github.com/apple/swift-distributed-tracing/blob/main/Sources/Instrumentation/Instrument.swift

We can set aside a moment to talk this through this week if you want @glbrntt? It'd be really valuable to use the tracing types here; The only holdoff from 1.0 them all is projects like this having a serious look at it, as well as the swift-log discussion we're having right now about propagation, so it'd be really a waste to invent your own "extractor" types right now.